### ISSUE TITLE
waitForView scrolls UITableView

### ISSUE THREAD

**Issue by blindjoedeath** [kif-framework/KIF#1316]
When I use method `KIFUITestActor.waitForView(withAccessibilityIdentifier:)` to find `UITableView` it gets scrolled. Has anyone encountered this?

**Comment by knellr** [comment]
Yes, I've encountered this. It was a long time ago that I looked into it, I seem to recall that there was some logic to reset it that didn't always reset it appropriately.

Since we were mixing snapshot testing with UI testing this was an issue for us, which we resolved in our tests by wrapping it in a function `waitForViewAndResetScrollPosition`. We had to do the same waiting for absence / tappable views.

**Comment by justinseanmartin** [comment]
@blindjoedeath - There is a property that can be set within `UIViewTestActor` (AKA `viewTester`) via [`.usingCurrentFrame()`](https://github.com/kif-framework/KIF/blob/master/Sources/KIF/Classes/KIFUIViewTestActor.m#L57-L61) that will prevent the view from changing scroll position. When this is set, the element being searched for must already be in the visible/tappable region of the screen to be matched.

This was an unfortunate side effect of some changes required due to an assertion that was added by Apple within UITableView and UICollectionView that prevents KIF from programmatically dequeueing offscreen elements. KIF tries to scroll things back to the original view position if a matching element isn't found, but it does a recursive depth first search to find matching elements, potentially scrolling unnecessary views along the way.

Take a look at #1296 and #1307 for more context.

Does this solve your need?

**Comment by blindjoedeath** [comment]
Yes, thank you!
